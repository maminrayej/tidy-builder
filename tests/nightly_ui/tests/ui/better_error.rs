#![feature(rustc_attrs)]

#[derive(tidy_builder::Builder)]
struct Item {
    field1: Option<usize>,
    field2: usize,
}


fn main() {
    let item = Item::builder()
        .field1(10)
        .build();
}