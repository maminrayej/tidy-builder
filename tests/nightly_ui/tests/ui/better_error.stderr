error[E0277]: missing `field2`
  --> tests/ui/better_error.rs:13:10
   |
13 |         .build();
   |          ^^^^^ provide `field2` before calling `.build()`
   |
   = help: the trait `HasField2` is not implemented for `ItemBuilder<false>`
   = help: the trait `HasField2` is implemented for `ItemBuilder<true>`
note: required by a bound in `ItemBuilder::<P0>::build`
  --> tests/ui/better_error.rs:3:10
   |
3  | #[derive(tidy_builder::Builder)]
   |          ^^^^^^^^^^^^^^^^^^^^^ required by this bound in `ItemBuilder::<P0>::build`
   = note: this error originates in the derive macro `tidy_builder::Builder` (in Nightly builds, run with -Z macro-backtrace for more info)
